packages/flutter/lib/src/material/scrollbar.dart

```diff
@@ -2,236 +2,200 @@
+    radius: radius,
+    fadeDuration: _kScrollbarFadeDuration,
+    timeToFade: _kScrollbarTimeToFade,
+    pressDuration: Duration.zero,
+  );
+
+  /// Controls if the track will show on hover and remain, including during drag.
   ///
-  /// If this is non-null, it will be used as the fixed radius of the scrollbar
-  /// on all platforms, whether the scrollbar is being dragged by the user or
-  /// not. By default (if this is left null), each platform will get a radius
-  /// that matches the look and feel of the platform, and the radius may
-  /// change while the scrollbar is being dragged if the platform look and feel
-  /// calls for such behavior.
-  final Radius? radius;
+  /// Defaults to false, cannot be null.
+  final bool showTrackOnHover;
 
   @override
   _ScrollbarState createState() => _ScrollbarState();
 }
 
-class _ScrollbarState extends State<Scrollbar> with SingleTickerProviderStateMixin {
-  ScrollbarPainter? _materialPainter;
-  late TextDirection _textDirection;
-  late Color _themeColor;
-  late bool _useCupertinoScrollbar;
-  late AnimationController _fadeoutAnimationController;
-  late Animation<double> _fadeoutOpacityAnimation;
-  Timer? _fadeoutTimer;
+class _ScrollbarState extends RawScrollbarState<Scrollbar> {
+  late AnimationController _hoverAnimationController;
+  bool _dragIsActive = false;
+  bool _hoverIsActive = false;
+
+  Set<MaterialState> get _states => <MaterialState>{
+    if (_dragIsActive) MaterialState.dragged,
+    if (_hoverIsActive) MaterialState.hovered,
+  };
+
+  MaterialStateProperty<Color> get _thumbColor {
+    return MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.dragged))
+        return const Color(0xFF616161);
+
+      // If the track is visible, the thumb color hover animation is ignored and
+      // changes immediately.
+      if (states.contains(MaterialState.hovered) && widget.showTrackOnHover)
+        return const Color(0xFF757575);
+
+      return Color.lerp(
+        const Color(0xFFE0E0E0),
+        const Color(0xFF757575),
+        _hoverAnimationController.value,
+      )!;
+    });
+  }
+
+  MaterialStateProperty<Color> get _trackColor {
+    return MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.hovered) && widget.showTrackOnHover)
+        return const Color(0xFF424242).withOpacity(0.04);
+      return const Color(0x00000000);
+    });
+  }
+
+  MaterialStateProperty<Color> get _trackBorderColor {
+    return MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.hovered) && widget.showTrackOnHover)
+        return const Color(0xFFE0E0E0);
+      return const Color(0x00000000);
+    });
+  }
+
+  MaterialStateProperty<double> get _thickness {
+    return MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+      if (states.contains(MaterialState.hovered) && widget.showTrackOnHover)
```

Comment:
Line 28 says `The thickness of the track and scrollbar thumb will become larger when hovering unless overridden by [thickness].`
This description seems to be inconsistent with the code logic.